sdk/keyvault/Azure.Security.Keyvault.Keys/src/KeyClient.cs

```diff
@@ -0,0 +1,175 @@
+        {
+            await Task.CompletedTask;
+            throw new NotImplementedException();
+        }
+
+
+        public virtual Response<Key> CreateECKey(ECKey ecKey, CancellationToken cancellationToken = default)
+        {
+            throw new NotImplementedException();
+        }
+
+        public virtual async Task<Response<Key>> CreateECKeyAsync(ECKey ecKey, CancellationToken cancellationToken = default)
+        {
+            await Task.CompletedTask;
+            throw new NotImplementedException();
+        }
+
+        public virtual Response<Key> CreateRSAKey(RSAKey rsaKey, CancellationToken cancellationToken = default)
+        {
+            throw new NotImplementedException();
+        }
+
+        public virtual async Task<Response<Key>> CreateRSAKeyAsync(RSAKey rsaKey, CancellationToken cancellationToken = default)
+        {
+            await Task.CompletedTask;
+            throw new NotImplementedException();
+        }
+
+        public virtual Response<Key> Update(KeyBase key, IList<string> keyOps = null, CancellationToken cancellationToken = default)
+        {
+            throw new NotImplementedException();
+        }
+
+        public virtual async Task<Response<Key>> UpdateAsync(KeyBase key, IList<string> keyOps = null, CancellationToken cancellationToken = default)
+        {
+            await Task.CompletedTask;
+            throw new NotImplementedException();
+        }
+
+        public virtual Response<Key> Get(string name, string version = null, CancellationToken cancellationToken = default)
+        {
+            throw new NotImplementedException();
+        }
+
+        public virtual async Task<Response<Key>> GetAsync(string name, string version = null, CancellationToken cancellationToken = default)
+        {
+            await Task.CompletedTask;
+            throw new NotImplementedException();
+        }
+
+        public virtual Response<DeletedKey> GetDeletedKey(string name, CancellationToken cancellationToken = default)
+        {
+            throw new NotImplementedException();
+        }
+
+        public virtual async Task<Response<DeletedKey>> GetDeletedKeyAsync(string name, CancellationToken cancellationToken = default)
+        {
+            await Task.CompletedTask;
+            throw new NotImplementedException();
+        }
+
+        public virtual Response<DeletedKey> Delete(string name, CancellationToken cancellationToken = default)
+        {
+            throw new NotImplementedException();
+        }
+
+        public virtual async Task<Response<DeletedKey>> DeleteAsync(string name, CancellationToken cancellationToken = default)
+        {
+            await Task.CompletedTask;
+            throw new NotImplementedException();
+        }
+
+        public virtual IEnumerable<Response<DeletedKey>> GetBatchDeletedKeys(CancellationToken cancellationToken = default)
```

Comment:
>GetBatchDeletedKeys [](start = 57, length = 19)

For secrets we used the verbiage 'all' rather than 'batch', so this should be GetAllDeletedKeys

Reply:
For configuration service there was a discussion about it, and the conclusion was to use batch instead of all.
I'm going to renaming it and delete the batch and all

Reply:
In the case of the configuration service there is filtering, so you're not necessarily returning all, so batch makes sense.  This is not the case for key vault all entities will be returned always, hence all. 

---
In reply to: [287145821](https://github.com/Azure/azure-sdk-for-net/pull/6374#discussion_r287145821) [](ancestors = 287145821)